filesystem/flysystem/FlysystemAssetStore.php

```diff
@@ -110,6 +110,9 @@ public function setPublicFilesystem(Filesystem $filesystem) {
 	 * @return Filesystem
 	 */
 	public function getPublicFilesystem() {
+		if(!$this->publicFilesystem) {
+			throw new Exception("Filesystem misconfiguration error");
```

Comment:
We can even have our own exception classes! Imagine. 

But yes, let's not just throw around base `Exception`s